docs/understanding-preprocessing-codes.md

```diff
@@ -0,0 +1,490 @@
+
+The `layer_traza` parameter defines the land use data (Trazabilidad) with the Trazabilidad layer having four fields (T1, T2, T3, T4) with land use data for 4 distinct years. Initially, the LUC shapefile is read and a filter is used to identify polygons that went from non-forest to forest between T1 and T2. The year and name of the disturbance are assigned and a random year is assigned between T1 and T2 for the disturbance.
+
+The important columns are left out, with 2014, 2015 and 2016 years being included and the partial layers bind together. The substitution layer is further developed, from forest to exotic plantations. The polygons that went from forest to plantations between T1 and T2 are filtered and the year and type of substitution are assessed.
+
+```R
+sust_t2<- dplyr::filter(
+  traza,
+  T1 %in% c("0403", "040203", "040202", "040204", "040201", "0402","0304"),
+  T2 %in% c("0401"))
+
+sust_t2$year<-year_t2
+sust_t2$Perturb<-"Sustitucion"
+
+sust_t2$year<-sample(year_t1:(year_t2),nrow(sust_t2),replace=TRUE)
+
+sust_t2<-sust_t2[,c("year","Perturb")]
+```
+
+The disturbance year is randomized, along with disturbances from T3 to T4. The partial layers are bind together and the forest loss disturbances layer is bound together. We add the project to the WGS84 latitude and longitude and finally write the shapefile.
+
+### Forest Degradation Layer
+
+The Forest Degradation layer code creates the degradation layers and disturbance matrix values, including the conservation areas. The parameters `input_pf` and `output_gcbm` define the folders where the permanent forest files are and the folder of the output data for the model.
+
+The other parameters include the permanent forest layer in the FREL for two years, 2001 and 2010, defined as `year_t1` and `year_t2`. The number of quantiles to represent the degradation is defined as `n_quantiles`.
+
+We read the permanent forest shapefile and the codes that indicate degradation are 2,3,4 and 10000, 10001 in the "Carta". Do note that the change in CO2 (CAM_CO2) has to be negative.
+
+```R
+pf <- st_read(dsn = input_pf, layer = layer_pf)
+
+degradation <- dplyr::filter(
+  pf,
+  Carta %in% c("2", "3", "4", "10000", "10001"),
+  CAM_CO2 < 0)
+```
+
+A random year is defined for the degradation between 2002 and 2010 and the regional average of CO2 is calculated. The percentage of the regional CO2 that is lost in each degradation pixel is also calculated and the shapefile is divided into two parts: degradation that occurs in conservation areas and degradation that occurs outside of conservation areas.
+
+```R
+degradation$year<-sample((year_t1+1):year_t2,nrow(degradation),replace=TRUE)
+degradation$year
+
+regional_CO2 <- (375.2900742 * 1.75 * 0.5)  * 0.5 * (44/12)
+
+degradation$p_regional <- (degradation$CAM_CO2 / regional_CO2) * (-1)
+
+min(degradation$CAM_CO2)
```

Comment:
Maybe this calls to the min and max can be removed from the docs, they where just for debugging purposes

Reply:
Done!